File: server/Domain/Models/Basket.cs
namespace Domain.Models
{
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;

    public class Basket : EntityBase
    {
        public Basket()
        {
            OrderLines = new List<OrderLine>();
        }

        [Required]
        public User User { get; set; }

        [Required]
        public ICollection<OrderLine> OrderLines { get; set; }

        [Required]
        public decimal Price { get; set; }
    }
}

File: client/src/interfaces/basket.ts
import { IOrderLine } from 'src/interfaces/orderLine';
import { IUser } from 'src/interfaces/user';

export interface IBasket {
  id: string;
  orderLines: IOrderLine[];
  price: number;
}

export interface IBasketProps {
  basket: IBasket;
}
